src/language-js/needs-parens.js

```diff
@@ -752,13 +752,8 @@ function needsParens(path, options) {
         typeof node.value === "number"
       );
 
-    case "AssignmentExpression": {
-      const grandParent = path.grandparent;
-
-      if (key === "body" && parent.type === "ArrowFunctionExpression") {
-        return true;
-      }
```

Comment:
This `AssignmentExpression` branch returns `true` by default. This check is not needed.